scorecard/profile_data/api_data.py

```diff
@@ -178,20 +178,25 @@ def muni_contact(self):
 
         return values
 
-    def audit_opinions(self):
-        values = []
+    def audit_opinions(self, last_opinion_year):
+        years = range(last_opinion_year-3, last_opinion_year+1)
+        opinions = {}
+
         for result in self.results["audit_opinions"]:
-            values.append(
-                {
-                    "date": result["financial_year_end.year"],
-                    "result": result["opinion.label"],
-                    "rating": result["opinion.code"],
-                    "report_url": result["opinion.report_url"],
+            opinions[result["financial_year_end.year"]] = {
+                "result": result["opinion.label"],
+                "rating": result["opinion.code"],
+                "report_url": result["opinion.report_url"],
+            }
+        for year in years:
+            if year not in opinions:
+                opinions[year] = {
+                    "result": None,
+                    "rating": None,
+                    "report_url": None,
                 }
```

Comment:
```suggestion
        for year in years:
            if year in opinions:
                values.append(opinions[year])
            else:
                values.append({
                    "result": None,
                    "rating": None,
                    "report_url": None,
                })
        return values
```

Reply:
this should simplify the change set

Reply:
and make it consistent with how the rest of the indicators work (array of years incl placeholders for missing data)

Reply:
Added these changes and reverted unnecessary changes to data format and demo-data